components/bl-data-grid-component/README.md

```diff
@@ -0,0 +1,30 @@
+| Filter            | *Checkbox*                                                                         | true          | Filter Logic            | YES          | YES        | This handler allows you to specify a filter option for all grid columns.                                                                |
+| Floating Filter   | *Checkbox*                                                                         | true          | Floating Filter Logic   | YES          | YES        | This handler allows you to specify whether or not the floating filter is displayed.                                                     |
+| Column Definition | *JSON*                                                                             |               | Column Definition Logic | YES          | YES        | This handler allows you to add columns to the component. Signature of column `{field: String, sortable: Boolean, filter: Boolean}`. |
+| Row Data          | *JSON*                                                                             |               | Row Data Logic          | YES          | YES        | This handler allows you to add data to the component. Watch [Codeless Examples](#Examples).                                              |
+| Height            | *Number*                                                                           | 500           | Height Logic            | YES          | YES        | This handler allows you to specify the height of the component.                                                                         |
+| Width             | *Number*                                                                           | 750           | Width Logic             | YES          | YES        | This handler allows you to specify the width of the component.                                                                          |
+| Theme             | *Select* <br> "alpine" \| "alpine-dark" \| "balham" \| "balham-dark" \| "material" | "alpine"      | Theme Logic             | YES          | YES        | This handler allows you to specify a component theme.                                                                                   |
+
+### Events
+
+| Name                       | Triggers                                     | Context Blocks                                                                 |
+|----------------------------|----------------------------------------------|--------------------------------------------------------------------------------|
+| On Cell Click Event        | when the user clicks a cell of grid          | Cell Params: `{key: value}`                                                    |
+
+## <a name="Examples"></a> Codeless Examples
+
+Adding data to the component:
+
+![markers example](./example-images/adding-data.png)
```

Comment:
does this `Row Data Logic` handler run on each render? if so, this is a bad sample because it will send a request to the server on each render which is bad practice